src/ImageSharp/Processing/Processors/Transforms/KernelMap.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Runtime.CompilerServices;
+
+using SixLabors.ImageSharp.Memory;
+using SixLabors.Memory;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Transforms
+{
+    /// <summary>
+    /// Holds the <see cref="ResizeKernel"/> values in an optimized contigous memory region.
+    /// </summary>
+    internal class KernelMap : IDisposable
+    {
+        private readonly Buffer2D<float> data;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KernelMap"/> class.
+        /// </summary>
+        /// <param name="memoryAllocator">The <see cref="MemoryAllocator"/> to use for allocations.</param>
+        /// <param name="destinationSize">The size of the destination window</param>
+        /// <param name="kernelRadius">The radius of the kernel</param>
+        public KernelMap(MemoryAllocator memoryAllocator, int destinationSize, float kernelRadius)
+        {
+            int width = (int)Math.Ceiling(kernelRadius * 2);
+            this.data = memoryAllocator.Allocate2D<float>(width, destinationSize, AllocationOptions.Clean);
```

Comment:
Good work on this. I'm too lazy at times to do the appropriate discovery work you did.